dali/kernels/slice/slice_gpu_test.cu

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <string>
+#include "dali/kernels/slice/slice_gpu.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/test_tensors.h"
+
+#define DEBUG_ENABLED 0
+#define DEBUG_OUTPUT \
+  if (!DEBUG_ENABLED) ; \
+  else std::cout // NOLINT
+
+namespace dali {
+namespace kernels {
+
+template <typename Container>
+std::string BatchToStr(const Container& batch, const std::string sample_prefix = "Sample ") {
+  std::stringstream ss;
+  for (int i = 0; i < batch.num_samples(); i++) {
+    ss << sample_prefix << i << ":";
+    for (auto& x : make_span(batch[i].data, batch[i].num_elements()))
+      ss << " " << static_cast<int64_t>(x);
+  }
+  return ss.str();
+}
+
+template <typename InputType_, typename OutputType_, std::size_t Dims_, std::size_t NumSamples_,
+          std::size_t DimSize_, typename SliceParamsGenerator_>
+struct SliceGPUTestArgs {
+  using InputType = InputType_;
+  using OutputType = OutputType_;
+  static constexpr std::size_t Dims = Dims_;
+  static constexpr std::size_t NumSamples = NumSamples_;
+  static constexpr std::size_t DimSize = DimSize_;
+  using SliceArgsGenerator = SliceParamsGenerator_;
+};
+
+template <typename ExpectedShape>
+void AssertExpectedDimensions(const TensorShape<>& tensor_shape,
+                              const ExpectedShape& expected_shape) {
+  for (int j = 0; j < tensor_shape.size(); j++) {
+    ASSERT_EQ(expected_shape[j], tensor_shape[j]);
+  }
+}
+
+template <typename TestArgs>
+class SliceGPUTest : public ::testing::Test {
+ public:
+  using InputType = typename TestArgs::InputType;
+  using OutputType = typename TestArgs::OutputType;
+  static constexpr std::size_t Dims = TestArgs::Dims;
+  static constexpr std::size_t NumSamples = TestArgs::NumSamples;
+  static constexpr std::size_t DimSize = TestArgs::DimSize;
+  using SliceArgsGenerator = typename TestArgs::SliceArgsGenerator;
+
+  void PrepareExpectedOutut(TestTensorList<InputType, Dims>& input_data,
```

Comment:
Typo: PrepareExpectedOutut -> PrepareExpectedOutput

Reply:
Good catch! Will fix in next PR